libs/screen/image.ts

```diff
@@ -118,6 +130,121 @@ namespace helpers {
         imageDrawLine(img, x + w, y + h, x, y + h, c)
     }
 
+    export function imageDrawCircle(img: Image, cx: number, cy: number, r: number, col: number) {
+        cx = cx | 0;
+        cy = cy | 0;
+        r = r | 0;
+        // short cuts
+        if (r < 0) 
+            return;
+        else if (r == 0) {
+            img.setPixel(cx, cy, col);
+            return;
+        } else if (r == 1) {
+            img.setPixel(cx + 1, cy, col);
+            img.setPixel(cx, cy + 1, col);
+            img.setPixel(cx - 1, cy, col);
+            img.setPixel(cx, cy - 1, col);
+            return;
+        }
+
+        const fcx = Fx8(cx);
+        const fcy = Fx8(cy);
+        const fr = Fx8(r);
+        const fr2 = Fx.leftShift(fr, 1);
+
+        let x = Fx.sub(fr, Fx.oneFx8)
+        let y = Fx.zeroFx8;
+        let dx = Fx.oneFx8;
```

Comment:
@pelikhan I just noticed oneFx8 is really 1/256 (i.e. 0.004) and twoFx8 is 2/256, which seems horribly wrong.

Is this what this code is expecting? Or does it just work 256 times slower than expected?

Reply:
256 slower